src/utils.js

```diff
@@ -249,16 +249,32 @@ export function orderProperties(properties, order) {
   if (!Array.isArray(order)) {
     return properties;
   }
-  if (order.length !== properties.length) {
-    throw new Error(
-      "uiSchema order list length should match object properties length");
+
+  const arrayToHash = arr => arr.reduce((prev, curr) => {
+    prev[curr] = true;
+    return prev;
```

Comment:
Changed. I wonder, though, if this has the same performance, since it copies the entire object every time now...